hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImageFormatProtobuf.java

```diff
@@ -172,13 +172,55 @@ public LoaderContext getLoaderContext() {
       return ctx;
     }
 
+    /***
+     * a thread for parallel MD5 computing to increase performance when loading
+     */
+    private static class DigestThread extends Thread {
+      volatile private IOException ioe = null;
+      volatile private MD5Hash digest = null;
+      private File file;
+
+      public DigestThread(File inFile) {
+        file = inFile;
+      }
+
+      public MD5Hash getDigest() {
+        return digest;
```

Comment:
What about checking if the exception is there and throw it if so?

Reply:
if digest have exception ，DigestThread have catch it.
try {
          digest = MD5FileUtils.computeMd5ForFile(file);
        } catch (IOException e) {
          ioe = e;
        } catch (Throwable t) {
          ioe = new IOException(t);
        }

Reply:
Correct, but we could move all the exception handling there. 
My proposal is that the getter here checks if ioe == null and throw it.
Then it goes to my other comment where you don't need to check for the exception anymore and just do a getter of the digest.